### PR TITLE
EW-828: Control option for "Spaltenboards"

### PR DESCRIPTION
**Pull request by mkreuzkam-cap**
# Short Description
- User can control exported column boards.
- Change density of checkboxes to compact.

## Links to Ticket and related Pull-Requests
- https://ticketsystem.dbildungscloud.de/browse/EW-828
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/4936/files

## Screenshots of UI changes
image

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/molecules/CommonCartridgeExportModal.vue [hunk 1]
```diff
@@ -190,11 +211,17 @@ const allTasksSelected = computed(() => {
 	return allTasks.value.every((topic) => topic.isSelected);
 });
 
+const allColumnBoards = ref<Array<Selection>>([]);
+const allColumnBoardsSelected = computed(() => {
+	return allColumnBoards.value.every((topic) => topic.isSelected);
```
Comment (psachmann) [inline src/components/molecules/CommonCartridgeExportModal.vue:216]: topic -> columnBoard in the arrow function

File: src/components/molecules/CommonCartridgeExportModal.vue [hunk 2]
```diff
@@ -190,11 +211,17 @@ const allTasksSelected = computed(() => {
 	return allTasks.value.every((topic) => topic.isSelected);
```
Comment (psachmann) [inline src/components/molecules/CommonCartridgeExportModal.vue:211]: topic -> task in the arrow function

File: src/components/molecules/CommonCartridgeExportModal.unit.ts [hunk 3]
```diff
@@ -168,4 +169,28 @@ describe("@/components/molecules/CommonCartridgeExportModal", () => {
...
+	describe("toggleAllColumnBoards", () => {
+		it("should start with true and change the value when click", async () => {
+			const wrapper = setup();
+			const nextBtn = wrapper.findComponent('[data-testid="dialog-next-btn"]');
+			await nextBtn.trigger("click");
+
+			const allColumnBoards = wrapper.findComponent(
+				'[data-testid="all-column-boards-checkbox"]'
+			);
+			expect(
```
Comment (psachmann) [inline src/components/molecules/CommonCartridgeExportModal.unit.ts:182]: Adding an empty line for better readability

File: src/components/molecules/CommonCartridgeExportModal.unit.ts [hunk 4]
```diff
@@ -168,4 +169,28 @@ describe("@/components/molecules/CommonCartridgeExportModal", () => {
...
+				allColumnBoards
+					.findAll("input")
+					.some((input) => input.attributes("value") === "true")
+			).toBe(true);
+
+			await allColumnBoards.trigger("click");
+			expect(
```
Comment (psachmann) [inline src/components/molecules/CommonCartridgeExportModal.unit.ts:189]: Adding an empty line for better readability

**Review by psachmann — COMMENTED** [review]
Manually tested on the dev deployment 👍 and some minor comments regarding better code readability

File: src/components/molecules/CommonCartridgeExportModal.vue (hunk 1 above)
Comment (mkreuzkam-cap) [reply src/components/molecules/CommonCartridgeExportModal.vue]: done

File: src/components/molecules/CommonCartridgeExportModal.unit.ts (hunk 3 above)
Comment (mkreuzkam-cap) [reply src/components/molecules/CommonCartridgeExportModal.unit.ts]: done

File: src/components/molecules/CommonCartridgeExportModal.vue (hunk 2 above)
Comment (mkreuzkam-cap) [reply src/components/molecules/CommonCartridgeExportModal.vue]: done

File: src/components/molecules/CommonCartridgeExportModal.unit.ts (hunk 4 above)
Comment (mkreuzkam-cap) [reply src/components/molecules/CommonCartridgeExportModal.unit.ts]: done

**Review by Fshmit — APPROVED** [review]

**Pull request merged**